### PR TITLE
classes from @doolse's purescript-records:

### PR DESCRIPTION
**Pull request by athanclark**
This is in regards to the following two issues:

https://github.com/doolse/purescript-records/pull/2#issuecomment-326206312
https://github.com/doolse/purescript-records/issues/1

### CODE REVIEW

File: src/Data/Record/Class.purs [hunk 1]
```diff
@@ -0,0 +1,51 @@
+module Data.Record.Class (
+    class Subrow,
+    class RecordMerge,
+    class IntersectRow,
+    merge,
+    unionMerge
+) where
+
+-- | Proof that row `r` is a subset of row `s`
+class Subrow (r :: # Type) (s :: # Type)
```
Comment (paf31) [inline src/Data/Record/Class.purs:10]: We should just use `Union` directly here.

File: src/Data/Record/Class.purs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+instance srInst :: Union r t s => Subrow r s
+
+-- | Proof of row `i` being the intersection of rows `ri` and `si`,
+-- | `r` is `i` subtracted from `ri` and
+-- | `s` is `i` subtracted from `si`
+class IntersectRow (ri :: # Type) (si :: # Type) (i :: # Type) (r :: # Type) (s :: # Type)
```
Comment (paf31) [inline src/Data/Record/Class.purs:16]: This really belongs in `typelevel-prelude` since it's not record specific.

**Comment by paf31** [comment]
Thanks! Could you please update the LICENSE file to refer to the original as well?

**Comment by athanclark** [comment]
@paf31 oh hm.. I don't think @doolse included a LICENSE in his original repo - https://github.com/doolse/purescript-records

However, if there was a license to be included - should it be inserted in the top of the Class.purs source file? I'm not sure what the usual routine would be for something like that.

**Comment by paf31** [comment]
We would usually add it to the LICENSE file in this repo (see the compiler license file as an example). This shouldn't apply for code which is PR'd by the owner of the original, though.

**Comment by athanclark** [comment]
@doolse could you sign off on this? @paf31 is hesitant on pulling this code, because it's unlicensed.

**Comment by paf31** [comment]
To be clear, I'd prefer if the original repo had a LICENSE file. 

I'd like to discuss the content of the PR a bit as well though. It's not obvious to me that this does the right thing in the presence of duplicate labels. I wonder if something using `RowList` would be conceptually simpler, with possibly similar inference properties too.

**Pull request closed without merging**